src/main/java/mezz/jei/util/AnnotatedInstanceUtil.java

```diff
@@ -28,7 +30,7 @@ private AnnotatedInstanceUtil() {
 	private static <T> List<T> getInstances(Class<?> annotationClass, Class<T> instanceClass) {
 		Type annotationType = Type.getType(annotationClass);
 		List<ModFileScanData> allScanData = ModList.get().getAllScanData();
-		List<String> pluginClassNames = new ArrayList<>();
+		Set<String> pluginClassNames = new HashSet<>();
```

Comment:
This is going to ultimately change the order that plugins load, possibly exposing some issues in mod plugins that have undeclared dependencies on each other.
Can you use a `LinkedHashSet` here to ensure the order stays the same?